### PR TITLE
feat: create squeeze page record

### PR DESCRIPTION
**Pull request by coderoyalty**
## Description: What this PR is.

This PR is about creating a squeeze page record.

## How it can be tested.

Make a `POST` request to the endpoint `/api/v1/squeeze` with a request body matching this:

```json
{
  "email": "user@example.com",
  "first_name": "John",
  "last_name": "Doe",
  "phone": "08098761234",
  "location": "Lagos, Nigeria",
  "job_title": "Software Engineer",
  "company": "X-Corp",
  "interests": ["Web Development", "Cloud Computing"],
  "referral_source": "LinkedIn"
}
```

Response:

```json
{
  "message": "Your request has been received. You will get a template shortly.",
  "status_code": 201
}
```

## Checklist of What I did

- [x] validate and verify incoming user data
- [x] created a record for the squeeze page in the database
- [x] I returned the appropriate response
- [x] I handle errors and return appropriate status codes
- [x] added unit tests for the squeeze controller and services
- [x] provided documentation for this feature
- [ ] send the template to a specified email address using an email service (e.g., SendGrid, SMTP).

## Reference Issue
#80: [FEAT]: Implement Backend API for Squeeze Page - Backend

### CODE REVIEW

**Review by davidolanrewaju — APPROVED** [review]

**Comment by Heba-WebDev** [comment]
I released this service today and assigned it to NickJrDev. If he asks for a pull request before the deadline, the endpoint is his. I'm just letting you know.

**Review by rugue — APPROVED** [review]

**Comment by DevNickJr** [comment]
> I released this service today and assigned it to NickJrDev. If he asks for a pull request before the deadline, the endpoint is his. I'm just letting you know.

Thank you very much. And yes, I already made a pull request. Pull 400, 8mins before this guy. Still awaiting reviews

**Comment by coderoyalty** [comment]
> 8mins before this guy.

https://github.com/hngprojects/hng_boilerplate_nestjs/pull/398
> this guy

Name: Emmanuel

**Comment by Heba-WebDev** [comment]
@DevNickJr @coderoyalty We'll solve this okay. No need for these comments please.

File: src/modules/squeeze/squeeze.controller.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { skipAuth } from '../../helpers/skipAuth';
+import { ApiCreatedResponse, ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+
+@ApiTags('squeeze')
+@Controller('squeeze')
+export class SqueezeController {
+  constructor(private readonly squeezeService: SqueezeService) {}
+
+  @skipAuth()
+  @Post()
```
Comment (Homoakin619) [inline src/modules/squeeze/squeeze.controller.ts:13]: Is this route protected ?

File: src/modules/squeeze/squeeze.service.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { Squeeze } from './entities/squeeze.entity';
+
+@Injectable()
+export class SqueezeService {
+  constructor(
+    @InjectRepository(Squeeze)
+    private squeezeRepository: Repository<Squeeze>
+  ) {}
+
+  async createSqueeze(createSqueezeDto: CreateSqueezeDto) {
```
Comment (Homoakin619) [inline src/modules/squeeze/squeeze.service.ts:14]: this method was also implemented in pull request #400 , seems you guys have duplicate tickets

**Comment by Homoakin619** [comment]
It appears you kinda have a duplicate ticket with #400  you both should sync together avoid duplication of codes, the pr has almost everything as you do have here

File: src/modules/squeeze/squeeze.controller.ts (hunk 1 above)
Comment (coderoyalty) [reply src/modules/squeeze/squeeze.controller.ts:13]: the route is not duplicated.

**Comment by Heba-WebDev** [comment]
@coderoyalty has been assigned another endpoint within the squeeze service. Please this can't be merged. The endpoint belongs to @DevNickJr. I'll close this myself when @coderoyalty opens another pull request for his new endpoint

**Pull request closed without merging**